### PR TITLE
Feature/search and filter

### LINKED ISSUE #3: As a user, I should be able to filter by product type

### LINKED ISSUE #4: As a user, I should be able to search for makeup by brand
- Discuss making an API call using their API they provide for this or setting all makeup items to state and parsing though that instead of making an API call for every keyboard stroke during a search

### PR DESCRIPTION
**Pull request by lauraguerra1**
What I did:
- Create a search by brand name feature
- create a filter by product type feature 
- add minimal styling 
- Removed form component and moved the functionality into the NavBar component so that the items could live the same flex container 
- Added a Link to favorites path (this path is not yet defined in our routes, no functionality here yet) 
- closes #3 
- closes #4 

Testing Notes/Extras: 
- there is no current testing to date - I am 'testing' the functionality via a console log of the filtered items every time the state changes, but once we use this functionality to actually display items, we will need to incorporate cypress testing 

Did this break anything?

- [x]  No
- [ ]  Yes

Type of change

- [x]  New feature (non-breaking change which adds functionality)
- [ ]  Bug fix (non-breaking change which fixes an issue)
- [ ]  Refactor(DRY-ing up/ reorganizing code, etc.)
- [ ]  Super small fix (Corrected a typo, removed a comment, etc.)
- [ ]  Skip all the other stuff and briefly explain the fix.

Checklist:

- [ ]  If this code needs to be tested, all tests are passing
- [x]  The code runs locally
- [x]  The code is DRY. If it's not, I tried my best
- [x]  I reviewed my code before pushing

What's next?
- Add product cards to display the results of the filter/search functionality 
- Cypress testing for this feature

### CODE REVIEW

**Review by fernandorobles97 — APPROVED** [review]
Looks great so far! Good job 👍🏼

File: src/apiTypes.tsx
Comment (paulina-isabel) [inline src/apiTypes.tsx:1]: Good idea.

File: src/components/App/App.tsx
Comment (paulina-isabel) [inline src/components/App/App.tsx:1]: Love the loader & conditional render.

**Review by paulina-isabel — APPROVED** [review]
Looking good!

File: src/apiCalls.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { Product } from "./apiTypes"
+
+const handleError = (response: Response): Promise<Product[]> => {
+  if(!response.ok) {
```
Comment (hfaerber) [inline src/apiCalls.tsx:4]: Nice response checking and error throwing here!

File: src/components/App/App.css [hunk 2]
```diff
@@ -1,35 +1,31 @@
...
+    animation: logo-spin infinite 2s linear;
+  }
 }
 
-.App-link {
-  color: #61dafb;
-}
 
-@keyframes App-logo-spin {
+@keyframes logo-spin {
```
Comment (hfaerber) [inline src/components/App/App.css:28]: Love seeing some animation!

File: src/components/App/App.tsx [hunk 3]
```diff
@@ -1,18 +1,67 @@
...
+    }))
+  }
+
+  useEffect(() => {
+    const apiCall = async () => {
+      setLoading(true)
+      try {
+        setAllProducts(await getAllProducts())
+        setLoading(false)
+      }catch(error) {
```
Comment (hfaerber) [inline src/components/App/App.tsx:32]: Nice application of a catch block here

File: src/components/App/App.tsx [hunk 4]
```diff
@@ -1,18 +1,67 @@
...
+      <NavBar loading={loading} products={allProducts} updateProducts={updateFilteredProducts}/>
+      {loading ? 
+        <section className='loading-container'>
+          <img className='loading' src={logo} alt='Makeup 360 spinning logo' />
+          <p className='loading-txt'>Loading...</p> 
+        </section>
+        : 
+        <Routes>
+          <Route path='/' element={<ProductContainer filteredProducts={filteredProducts}/>} />
+          <Route path='/product/:id' element={<></>} />
```
Comment (hfaerber) [inline src/components/App/App.tsx:62]: You will want to make sure you have a * route for handling bad routes. You should be able to check that bad routes are handled properly by checking "localhost:3000/nonsense" and "localhost:3000/product/someId/nonsense". Users should see messaging like a 404 telling them that page doesn't exist and should have a way to navigate back to the homepage.

File: src/components/NavBar/NavBar.tsx [hunk 5]
```diff
@@ -1,11 +1,53 @@
...
+  const updateCategory = (category: string | null):void => {
+    setSelectedCategory(prevState => {
+      return prevState !== category ? category : null
+    })
+  }
+
+  useEffect(() => {
+    searchData || selectedCategory ? setClearAllowed(true) : setClearAllowed(false)
+    updateProducts(searchData, makeSnakeCase(selectedCategory))
+  }, [searchData, selectedCategory])
```
Comment (hfaerber) [inline src/components/NavBar/NavBar.tsx:31]: Love seeing the dependency array used in this way!

File: src/helpers.tsx [hunk 6]
```diff
@@ -0,0 +1,3 @@
+const makeSnakeCase = (words: string | null) => words ? words.replaceAll(' ', '_').toLowerCase() : null
```
Comment (hfaerber) [inline src/helpers.tsx:1]: Love the use of this helpers file!

**Review by hfaerber — COMMENTED** [review]
Sorry for the late PR review.  Got stuck on zooms all day Thurs and Fri.  This all looks like really good work y'all.  Excited to see the final product!

**Pull request merged**